openpype/hosts/houdini/plugins/publish/validate_frame_range.py

```diff
@@ -0,0 +1,75 @@
+# -*- coding: utf-8 -*-
+import pyblish.api
+from openpype.pipeline import PublishValidationError
+from openpype.pipeline.publish import RepairAction
+from openpype.hosts.houdini.api.action import SelectInvalidAction
+
+import hou
+
+
+class HotFixAction(RepairAction):
+    """Set End frame to the minimum valid value."""
+
+    label = "End frame hotfix"
+
+
+class ValidateFrameRange(pyblish.api.InstancePlugin):
+    """Validate Frame Range.
+
+    Due to the usage of start and end handles,
+    then Frame Range must be >= (start handle + end handle)
+    which results that frameEnd be smaller than frameStart
+    """
+
+    order = pyblish.api.ValidatorOrder - 0.1
+    hosts = ["houdini"]
+    label = "Validate Frame Range"
+    actions = [HotFixAction, SelectInvalidAction]
+
+    def process(self, instance):
+
+        invalid = self.get_invalid(instance)
+        if invalid:
+            nodes = [n.path() for n in invalid]
+            raise PublishValidationError(
+                "Invalid Frame Range on: {0}".format(nodes),
+                title="Invalid Frame Range"
+            )
+
+    @classmethod
+    def get_invalid(cls, instance):
+
+        if not instance.data.get("instance_node"):
+            return
+
+        rop_node = hou.node(instance.data["instance_node"])
+        if instance.data["frameStart"] > instance.data["frameEnd"]:
+            cls.log.error(
+                "Wrong frame range, please consider handle start and end.\n"
+                "frameEnd should at least be {}.\n"
+                "Use \"End frame hotfix\" action to do that."
+                .format(
+                    instance.data["handleEnd"] +
+                    instance.data["handleStart"] +
+                    instance.data["frameStartHandle"]
+                )
+            )
+            return [rop_node]
+
+    @classmethod
+    def repair(cls, instance):
+        rop_node = hou.node(instance.data["instance_node"])
+
+        frame_start = int(instance.data["frameStartHandle"])
+        frame_end = int(
+            instance.data["frameStartHandle"] +
+            instance.data["handleStart"] +
+            instance.data["handleEnd"]
+        )
+
+        if rop_node.parm("f2").rawValue() == "$FEND":
+            hou.playbar.setFrameRange(frame_start, frame_end)
+            hou.playbar.setPlaybackRange(frame_start, frame_end)
+            hou.setFrame(frame_start)
+        else:
+            rop_node.parm("f2").set(frame_end)
```

Comment:
I still feel the only sensible 'fix' is to just disable asset handles - why wouldn't we add a Repair Action that just disables the "Use asset data" toggle on the instance, the repair could be named: "Disable using asset handles" or whatever.

The only other _hotfix_ that I think makes somewhat sense is that if the user didn't know he had to include handles to begin with - then the repair label could be "Add handles to start and end range" and we'd actually add them so that the handles are then inclusive to the frame range.

What do you think?

Moving just the end frame feels so weird.

Reply:
`Disable using asset handles` makes sense but I don't know if a validator can access an attribute def of a collector ? 
`Add handles to start and end range` This one is much simpler but it's inclusive to the frame range! 

maybe should I just remove this repair action ?